lib/Connection.php

```diff
@@ -335,6 +344,18 @@ public function query($sql, &$values=array())
 				throw new DatabaseException($this);
 		} catch (PDOException $e) {
 			throw new DatabaseException($e);
+		} finally {
```

Comment:
Could make it work without finally, would just have to catch and rethrow all exceptions in a default catch block and they get the "end time" in all three parts of the try/catch.